content/docs/5.1/concepts/explore-tikv-features/fault-tolerance.md

```diff
@@ -1,10 +1,201 @@
 ---
-title: Fault tolerance
-description: TODO
+title: Fault tolerance and recovery
+description: Fault tolerance, recovery
 menu:
     "5.1":
         parent: Features
         weight: 2
 ---
 
-TODO
+This page walks you through a simple demonstration of how TiKV remains available during and recovers after, failure.
+
+1. Starting with a 6-node local cluster with the default 3-way replication.
+2. Run a sample workload via [go-ycsb](https://github.com/pingcap/go-ycsb), terminate a node to simulate a failure, and see how the cluster continues uninterrupted.
+3. Leave that node offline for long enough to watch the cluster repair itself by re-replicating missing data to other nodes.
+4. Prepare the cluster for 2 simultaneous node failures by increasing to 5-way replication, then take two nodes offline at the same time, and again see how the cluster continues uninterrupted.
+
+
+## Prerequisites
+
+1. Installed [TiUP](https://github.com/pingcap/tiup) version **v1.5.2** or above as described in [TiKV in 5 Minutes](../../tikv-in-5-minutes) for cluster deployment
+2. Clone and compile tool [go-ycsb](https://github.com/pingcap/go-ycsb) in local for launch a workload
+3. Install [client-py](https://github.com/tikv/client-py) to interact with the TiKV cluster.
+
+
+## Step 1. Start a 6-node cluster
+
+With a new terminal session, use `tiup playground` command to launch a 6-node local cluster.
+```sh
+tiup playground --mode tikv-slim --kv 6
+```
+This command will give you a hint about components' addresses. It will be used in the following steps:
+```txt
+Playground Bootstrapping...
+Start pd instance
+Start tikv instance
+Start tikv instance
+Start tikv instance
+Start tikv instance
+Start tikv instance
+Start tikv instance
+PD client endpoints: [127.0.0.1:2379]
+To view the Prometheus: http://127.0.0.1:44549
+To view the Grafana: http://127.0.0.1:3000
+```
+
+**By now, each region in this cluster will have 3 replicas according to the default configuration.**
+
+## Step 2. Write data
+
+On another terminal session, use [go-ycsb](https://github.com/pingcap/go-ycsb) to launch a workload.
+
+1. Clone the `go-ycsb` from GitHub.
+    ```sh
+    git clone https://github.com/pingcap/go-ycsb.git
+    ```
+2. Build the application from the source.
+    ```sh
+    make
+    ```
+3. Load a workload using `go-ycsb` with **10000** keys.
+    ```sh
+    ./bin/go-ycsb load tikv -P workloads/workloada -p tikv.pd="127.0.0.1:2379" -p tikv.type="raw" -p recordcount=1000000
+    ```
+    This command will output the following content:
+    ```txt
```

Comment:
leave an empty line between normal text and code

Reply:
Noted.